src/commons/api/roas.rs

```diff
@@ -280,22 +286,186 @@ impl Ord for RoaDefinition {
     }
 }
 
-impl PartialOrd for RoaDefinition {
+impl PartialOrd for RoaPayload {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         Some(self.cmp(other))
     }
 }
 
-impl AsRef<TypedPrefix> for RoaDefinition {
+impl AsRef<TypedPrefix> for RoaPayload {
     fn as_ref(&self) -> &TypedPrefix {
         &self.prefix
     }
 }
 
+//------------ RoaConfiguration --------------------------------------------
+
+/// This type defines a saved RoaConfiguration. It includes the actual ROA
+/// payload that needs be authorized in a ROA object as well as other
+/// information that is only visible to users - i.e. an optional comment
+/// field, and in future perhaps other things such as tags used for
+/// classification/monitoring/bpp analysis.
+///
+/// Note that this type is backward compatible with the RoaDefinition type
+/// used until Krill 0.10.0.
```

Comment:
Is "is backward compatible" strong enough or is the case that this MUST remain backward compatible?

Reply:
yes! But I explained it better now (I think).

Essentially - if we can avoid upgrades and breaking changes we probably should :D